SlavePool.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace PHPPM;
+
+use React\Socket\ConnectionInterface;
+
+/**
+ * SlavePool singleton is responsible for maintaining a pool of slave instances
+ */
+class SlavePool
+{
+    private static $instance;
```

Comment:
Please no singleton :) Instantiating it in Master manually and hold it should be enough.